packages/replay-next/src/hooks/useDebounce.ts

```diff
@@ -1,11 +1,13 @@
-import { useEffect, useState } from "react";
+import { startTransition, useEffect, useState } from "react";
 
-export function useDebounce<Value>(value: Value, delay: number = 100): Value {
+export function useDebounce<Value>(value: Value, delay: number = 100) {
   const [debouncedValue, setDebouncedValue] = useState(value);
 
   useEffect(() => {
     const handler = setTimeout(() => {
-      setDebouncedValue(value);
+      startTransition(() => {
+        setDebouncedValue(value);
+      });
```

Comment:
I was on the fence about having a `useDebounce` hook automatically wrap the update in a transition, but I guess this change is okay. There's a good chance that we're debouncing a value _to_ Suspend with it, so I guess it's fine :)